### ISSUE TITLE
New examples requested

### ISSUE THREAD

**Issue by msaroufim** [pytorch/examples#1131]
Hi everyone, @svekars and I are looking to increase the number of new contributions to pytorch/examples, this might be especially interesting to you if you've never contributed to an open source project before.

At a high level, we're looking for new interesting models.

So here's what you need to do
1. Check out our contributing guide: https://github.com/pytorch/examples/blob/main/CONTRIBUTING.md
2. Pick a model idea - I've listed a few below, comment on this task so others know you're working on it
3. Implement your model from scratch using PyTorch, no external dependencies will be allowed to keep the examples as educational as possible

Your implementation needs to include
1. A folder with your code which needs to define
  1. Your model architecture
  2. Training code
  4. Evaluation code 
  5. An argparser 
3. Make sure your script runs in CI so it doesn't break in the future by adding it to `run_python_examples.sh`
4. README describing any usage instructions

As an example this recent contribution by @sudomaze is a good one to follow https://github.com/pytorch/examples/pull/1003/files

Here are some model ideas

## Model ideas

* [ ] Controlnet - Guided diffusion
* [ ] NERF 
* [x] Graph Neural Network @JoseLuisC99 
* [ ] Diffusion Model, stable diffusion or any variant of the architecture you like
* [x] Vision Transformer
* [ ] Video model
* [ ] Toolformer
* [ ] Differentiable physics
* [ ] Flownet
* [ ] Dreamfusion or any 3d model
* [ ] Language Translation
* [ ] Swin transformer

But I'm quite open to anything we don't have that's cool

**Comment by niyarrbarman** [comment]
Hey! I'm interested in contributing to the vision transformer model, but I don't have any prior open source contribution experience. Would it be okay for me to proceed with this project?

**Comment by msaroufim** [comment]
Yes please go for it

**Comment by xorsuyash** [comment]
@msaroufim I would Like to contribute to examples related to graph neural networks ,Is there any specific Dataset i should choose for this or i can choose any dataset of my own choice for examples .

**Comment by msaroufim** [comment]
Yes to all of the above

**Comment by aditikhare007** [comment]
@msaroufim I am interested for PyTorch Open-Source Contribution. Thank you for sharing New examples requested Note, would like to contribute to Diffusion Model, stable diffusion and Vision Transformer sections will keep you posted as my work progress.

**Comment by aditikhare007** [comment]
@msaroufim I am interested for PyTorch Open-Source Contribution. Thank you for sharing New examples requested Note, would like to contribute to Diffusion Model, stable diffusion and Vision Transformer sections will keep you posted as my work progress.
Please let me know your thoughts on taking up this project. Thank you.
Regards,
Aditi

**Comment by Krish2002** [comment]
hey!  I would love to contribute to Stable diffusion. Can i take this up ?

**Comment by msaroufim** [comment]
Hi @Krish2002 yes please go for it

**Comment by IMvision12** [comment]
I would love to contribute to FlowNet. Can I take this up?

**Comment by msaroufim** [comment]
Hi @IMvision12 please do!

**Comment by abhi-glitchhg** [comment]
I would like to add a video vision transformer model.

**Comment by msaroufim** [comment]
@abhi-glitchhg please do just keep in mind the implementation has to be from scratch and not just call the torcvision constructor

**Comment by JoseLuisC99** [comment]
I would like to contribute to Graph Neural Network. However, is there some specific task or model in mind or can I choose any?

**Comment by HemanthSai7** [comment]
Hi @msaroufim I would like to implement language translation using encoder-decoder architecture. Can I take this?

**Comment by msaroufim** [comment]
@JoseLuisC99 any task or model you like! As long as it's from scratch in pure Pytorch

**Comment by msaroufim** [comment]
@guptaaryan16 @HemanthSai7 @JoseLuisC99 @abhi-glitchhg assigned some models to you, lemme know if you need any help to get it over the finish line. Thanks!

**Comment by arunppsg** [comment]
Can I take up implementing Controlnet - guided diffusion?

**Comment by HemanthSai7** [comment]
@msaroufim can I use the transformers library to use the tokenizers?

**Comment by bhavyashahh** [comment]
@msaroufim can I take up NERF?

**Comment by QasimKhan5x** [comment]
@msaroufim I'd like to implement a text to 3d model. At the moment, I'm deciding between Test2Mesh and CLIP-Forge. DreamFusion seems a little complex.

**Comment by msaroufim** [comment]
@HemanthSai7 yes but the model should be pure PyTorch, bonus point for a from scratch tokenizer!
@bhavyashahh Sure!
@QasimKhan5x Sounds good, either of those models work

**Comment by svekars** [comment]
Please reference this issue in your PRs, like "Re #1131".

**Comment by IMvision12** [comment]
@msaroufim Sorry, but owing to university examinations I will not be able to participate this time. However, if anybody wants to take FlowNet, they are welcome to do so. :)

**Comment by HemanthSai7** [comment]
Wanted to give updates on my task. I have completed preparing the dataset(tokenization, data-loading, etc) for the translation task and will start with Positional Embeddings and other layers.

**Comment by msaroufim** [comment]
Thanks @HemanthSai7 

@yishengpei vision transformer was completed already but would be happy to review swin transformer

**Comment by HemanthSai7** [comment]
I'm seeing a lot of nan values when I print the attn_scores in nn.MultiheadAttention. Is it expected or is it due to a fault in the logic?

**Comment by HemanthSai7** [comment]
```python
def generate_square_subsequent_mask(seq_len):
  mask = (torch.triu(torch.ones(seq_len, seq_len)) == 1).transpose(0, 1)
  mask = mask.float().masked_fill(mask == 0, float('-inf')).masked_fill(mask== 1, float(0.0))
  return mask
  ```
  I'm unable to understand this way of generating masks used in the source code.

**Comment by msaroufim** [comment]
> I'm seeing a lot of nan values when I print the attn_output_weights in nn.MultiheadAttention in the decoder block. Is it expected or is it due to a fault in the logic?

Could you please share a repro? That's certainly not expected

**Comment by msaroufim** [comment]
> ```python
> def generate_square_subsequent_mask(seq_len):
>   mask = (torch.triu(torch.ones(seq_len, seq_len)) == 1).transpose(0, 1)
>   mask = mask.float().masked_fill(mask == 0, float('-inf')).masked_fill(mask== 1, float(0.0))
>   return mask
> ```
> 
> I'm unable to understand this way of generating masks used in the source code.

What exactly is confusing you?

Running it prints a reasonable looking mask to me

```pyhon
generate_square_subsequent_mask(10)
tensor([[0., -inf, -inf, -inf, -inf, -inf, -inf, -inf, -inf, -inf],
        [0., 0., -inf, -inf, -inf, -inf, -inf, -inf, -inf, -inf],
        [0., 0., 0., -inf, -inf, -inf, -inf, -inf, -inf, -inf],
        [0., 0., 0., 0., -inf, -inf, -inf, -inf, -inf, -inf],
        [0., 0., 0., 0., 0., -inf, -inf, -inf, -inf, -inf],
        [0., 0., 0., 0., 0., 0., -inf, -inf, -inf, -inf],
        [0., 0., 0., 0., 0., 0., 0., -inf, -inf, -inf],
        [0., 0., 0., 0., 0., 0., 0., 0., -inf, -inf],
        [0., 0., 0., 0., 0., 0., 0., 0., 0., -inf],
        [0., 0., 0., 0., 0., 0., 0., 0., 0., 0.]])
```

**Comment by HemanthSai7** [comment]
@msaroufim I was trying to create my own mask rule. Found out where I went wrong and fixed it.

**Comment by Potato-Cracker** [comment]
@msaroufim I would like to contribute to the OpenAI whisper implementation. Can I take this up?

**Comment by msaroufim** [comment]
Sure yeah! That sound cool

**Comment by Potato-Cracker** [comment]
What is the expected outcome in this context? The model and its pretrained weights can be obtained from OpenAI's Whisper repository. One possible approach is to utilize the model class and showcase the process of training or fine-tuning, as the original repository does not include training functionality.

**Comment by ebrahimpichka** [comment]
Hi, I see the GNN example is taken care of as a GCN example. Would other GNN variants such as GAT, GraphSage, etc. be helpful, or would they be counted as kinda duplicates?

**Comment by msaroufim** [comment]
@ebrahimpichka id love to see variants.

**Comment by ebrahimpichka** [comment]
@msaroufim great, I'd like to work on it.

**Comment by mingxzhao** [comment]
@msaroufim if no one is working on Flownet, can I be assigned?

**Comment by msaroufim** [comment]
@mingxzhao it's yours

**Comment by JoseLuisC99** [comment]
When you say "Differentiable physics", do you mean Physic-informed deep learning?  Or do you have some other architecture in mind?

**Comment by msaroufim** [comment]
Yeah that's what I had in mind but open to other cool sounding models

**Comment by JoseLuisC99** [comment]
Cool! I think that I can try PINNs and another modern solution using graph nets, so please assign me this problem.

**Comment by kausthub-kannan** [comment]
Hello, I am new to contributing to PyTorch, Can I try contributing a new example "Time Series Anamoly detection using LSTM Auto-encoders"? I did see an example for predicting time sequence, can we try anomaly detection?

Thank You

**Comment by msaroufim** [comment]
Hi folks yeah please go for it, no need to ask me for permission just send a PR and tag me so I can review

**Comment by sarthak247** [comment]
Greetings, 
I just came across this and would also like to be a part of it. As of now, I can only see that toolformer is the only model that is not taken up or assigned to anyone. Is it fine if I work on this? And also, is yes, I would need some assistance as to where to start from on this.
Regards,
Sarthak <3

**Comment by NoahSchiro** [comment]
Hey @HemanthSai7 and @msaroufim I've just submitted a PR for the language translation example. Sorry if I duplicated any work you've been working on but I have not heard movement on the language translation example for 6 months so I thought it would be safe to work on it.

**Comment by AbhiLegend** [comment]
Can I contribute an example of Drug Discovery using Pytorch with the RDKIT library?

**Comment by AbhiLegend** [comment]
> Can I contribute an example of Drug Discovery using Pytorch with the RDKIT library?

I would like to contribute this project on Pytorch this is currently with IPEX 
https://devpost.com/software/drug-discovery-using-amx-and-openvino

**Comment by msaroufim** [comment]
@AbhiLegend for this repo our goal is to have minimal implementations that are torch native as in they wouldn't use external libraries, for what you have in mind perhaps https://github.com/pytorch/tutorials is a better fit

**Comment by AbhiLegend** [comment]
> [@AbhiLegend](https://github.com/AbhiLegend) for this repo our goal is to have minimal implementations that are torch native as in they wouldn't use external libraries, for what you have in mind perhaps https://github.com/pytorch/tutorials is a better fit

Ok no problem I would like to contribute an example in the Differentiable Physics category. Can I go for it?

**Comment by msaroufim** [comment]
Yep go for it!

**Comment by AbhiLegend** [comment]
> Yep go for it!

@msaroufim I have added the example and opened a PR for it. #1332 . Can you review it?Thanks

**Comment by abhitorch81** [comment]
> > Yep go for it!
> 
> [@msaroufim](https://github.com/msaroufim) I have added the example and opened a PR for it. [#1332](https://github.com/pytorch/examples/pull/1332) . Can you review it?Thanks

My changes were reversed .Should I have do it once again? Any feedback will be appreciated.Thanks.

**Comment by RobertxPearce** [comment]
Working on a differentiable physics pendulum system identification example.

**Comment by msaroufim** [comment]
HI @RobertxPearce i wouldn't bother. I worry I haven't seen many high quality submissions mostly just due to people sending me straight up Claude generations. Goal here for me to was to have high quality submissions and in not enough in the weeds of unconcential architectures these days to do reviews in a reasonable timeframe 

I'll put some more thought tho into what might be a good contribution model for this repo and keep folks posted
